fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryTest.java

```diff
@@ -0,0 +1,416 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.fcrepo4;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+
+import org.fcrepo.client.FcrepoClient;
+import org.fcrepo.client.FcrepoResponse;
+import org.fcrepo.client.GetBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.vocabulary.RDF;
+
+import edu.unc.lib.dl.fedora.ObjectTypeMismatchException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.Fcrepo4Repository;
+import edu.unc.lib.dl.rdf.Premis;
+import edu.unc.lib.dl.util.URIUtil;
+
+/**
+ * 
+ * @author bbpennel
+ *
+ */
+public class RepositoryTest extends AbstractFedoraTest {
+
+	private Repository repository;
+
+	@Mock
+	private RepositoryObjectFactory objFactory;
+
+	@Mock
+	private FcrepoClient fcrepoClient;
+
+	@Before
+	public void init() {
+		initMocks(this);
+
+		repository = new Repository();
+		repository.setFedoraBase(FEDORA_BASE);
+		repository.setRepositoryObjectDataLoader(dataLoader);
+		repository.setRepositoryObjectFactory(objFactory);
+		repository.setClient(fcrepoClient);
+
+		PIDs.setRepository(repository);
+	}
+
+	@Test
+	public void mintDepositRecordPidTest() {
+		PID pid = repository.mintDepositRecordPid();
+
+		assertTrue(pid.getRepositoryPath().contains(RepositoryPathConstants.DEPOSIT_RECORD_BASE));
```

Comment:
Updated this to be a little bit more exact about where the DEPOSIT_RECORD_BASE is in the returned pid